### ISSUE TITLE
document functions in libgap-api.h, and related functions

### ISSUE THREAD

**Issue by dimpase** [gap-system/gap#2864]
as this is meant for the outside world, it need more info. E.g. the signature of 
`Obj GAP_EvalString(const char * cmd)` does not say what `Obj` will be. Here are few examples I tried (calling `ViewObj()` on each result:
```
GAP_EvalString("a := SymmetricGroup(4);") # ->
[ [ true, Sym( [ 1 .. 4 ] ), false,, "Sym( [ 1 .. 4 ] )" ] ]

GAP_EvalString("[1,2,3,4];9*[[42,7]];")          # ->
[ [ true, [ 1, 2, 3, 4 ], false,, "[ 1, 2, 3, 4 ]" ], 
  [ true, [ [ 378, 63 ] ], false,, "[ [ 378, 63 ] ]" ] ]

GAP_EvalString("[1,2,3,4];;")                          # ->
[ [ true, [ 1, 2, 3, 4 ], true,, "" ]]
```
One thing seems to be clear: it outputs the list of certain evaluations of the input,
one item per statement. 
I know (again, not from docs, but from experiments) that `ViewObj()` outputs `Obj` more or less as they are, without adding extra info, and thus each item is what `GAP_EvalString()` produces.
Each item is a list; its members are

1) a boolean, meaning unknown
2) something that looks like the string rep. of the result, regardless of the presense of `;;`
3) boolean, seems to indicate whether `;;` was present or not
4) ???
5) screen representation of the object (`;;` suppresses it)

Can this be clarified? I'd make a pull request with this in comments to the code...

**Comment by markuspf** [comment]
It executes `READ_ALL_COMMANDS` as "documented" here: https://github.com/gap-system/gap/blob/d7c32f314cd512fdd879dba22007626ea14ed8d8/src/streams.c#L85

note that `GAP_EvalString` is also just an example of what one *could* do; I am flexible as to what it should do/return/etc.

**Comment by dimpase** [comment]
Would it be correct to say that `GAP_EvalString` calls `READ_ALL_COMMANDS` on the input stream "provided" (how exactly?) by its argument?

**Comment by dimpase** [comment]
Further to `GAP_EvalString` I am using `ELM_LIST` and `LEN_LIST` in Sagemath [interface](https://trac.sagemath.org/ticket/22626). They are generally very useful, and ought to be exposed in API----actually, you already almost did this, putting `GAP_List_Length` etc in your `basic.c` test for libgap (I've shifted these functions to `common.c` in #2833)

**Comment by ChrisJefferson** [comment]
I agree it would better to give these functions some nice sensible names, like GAP_ElmList, GAP_LenList (feel free to use other names). These can of course just forward on to the standard GAP functions.

**Comment by fingolfin** [comment]
For a library ABI, it might be better to export ELM0_LIST instead of ELM_LIST, and leave it to the client code to handle with unbound entries; that way, there is one less place triggering an `ErrorQuit` in GAP to worry about.

**Comment by fingolfin** [comment]
This has been resolved in GAP 4.12 (by documenting `GAP_EvalString`, as ugly and bad as it is).
